packages/test-snaps/src/features/snaps/network-access/NetworkAccess.tsx

```diff
@@ -49,10 +72,41 @@ export const NetworkAccess: FunctionComponent = () => {
         id="sendNetworkAccessTest"
         className="mb-3"
         disabled={isLoading}
-        onClick={handleSubmit}
+        onClick={handleFetch}
       >
         Fetch
       </Button>
+
+      <br />
+
+      <ButtonGroup className="mb-3">
+        <Button
+          variant="primary"
+          id="startWebSocket"
+          disabled={isLoading}
+          onClick={handleStartWebSocket}
+        >
+          Start WebSocket
+        </Button>
+        <Button
+          variant="primary"
+          id="stopWebSocket"
+          disabled={isLoading}
+          onClick={handleStopWebSocket}
+        >
+          Stop WebSocket
+        </Button>
+
+        <Button
+          variant="primary"
+          id="getBlockNumber"
+          disabled={isLoading}
+          onClick={handleGetBlockNumber}
+        >
+          Get Block Number
+        </Button>
+      </ButtonGroup>
+
```

Comment:
For testing purposes it could also be nice to show the current connection state? So we listen to `open` and `close`, and add those to the state as well.

(Also makes me wonder if we should have non-persistent state)